x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -39,7 +39,7 @@ pub enum Error {
 }
 
 pub struct State<'a, K: Key> {
-    cache: &'a RefCell<HashMap<K, Vec<u8>>>,
+    cache: &'a RefCell<HashMap<K, Option<Vec<u8>>>>,
```

Comment:
The reason we have an `Option<Vec<u8>>` is to be able to know which is the data we know the state of (in the cache). As a guideline:
- `None`: We don't know the state of the data
- `Some(vec![])`: The data is has been deleted. The `get_host` is crucial to avoid deleting already deleted data (it would make the host error)
- `Some(data)`: There is some data